web-services/query/src/main/java/datawave/webservice/query/runner/BasicQueryBean.java

```diff
@@ -0,0 +1,384 @@
+                    }
+                    if (querySyntax.isEmpty()) {
+                        querySyntax.add("CUSTOM");
+                    }
+                    d.setQuerySyntax(querySyntax);
+                    
+                    break;
+                    
+                }
+            } catch (Exception e) {
+                log.error("Error setting query logic description", e);
+            }
+        }
+        
+        Principal p = ctx.getCallerPrincipal();
+        String authSting = AuthorizationsUtil.buildUserAuthorizationString(p);
+        response.setAuthString(authSting);
+        response.setTheQueryLogicDescription(theQld);
+        
+        return response;
+    }
+    
+    /**
+     * Display the query plan and link to basic query results for a simple query web UI in the quickstart
+     *
+     * @HTTP 200 Success
+     * @return datawave.webservice.result.QueryWizardStep3Response
+     * @RequestHeader X-ProxiedEntitiesChain use when proxying request for user, by specifying a chain of DNs of the identities to proxy
+     * @RequestHeader X-ProxiedIssuersChain required when using X-ProxiedEntitiesChain, specify one issuer DN per subject DN listed in X-ProxiedEntitiesChain
+     * @ResponseHeader X-OperationTimeInMS time spent on the server performing the operation, does not account for network or result serialization
+     */
+    @POST
+    @Produces({"application/xml", "text/xml", "application/json", "text/yaml", "text/x-yaml", "application/x-yaml", "text/html"})
+    @Path("/{logicName}/showQueryWizardStep3")
+    @GenerateQuerySessionId(cookieBasePath = "/DataWave/BasicQuery/")
+    @Interceptors({ResponseInterceptor.class})
+    @Timed(name = "dw.query.showQueryWizardStep3", absolute = true)
+    public QueryWizardStep3Response showQueryWizardStep3(@Required("logicName") @PathParam("logicName") String logicName,
+                    MultivaluedMap<String,String> queryParameters, @Context HttpHeaders httpHeaders) {
+        CreateQuerySessionIDFilter.QUERY_ID.set(null);
+        GenericResponse<String> createResponse;
+        QueryWizardStep3Response queryWizardStep3Response = new QueryWizardStep3Response();
+        try {
+            createResponse = queryExecutor.createQuery(logicName, queryParameters, httpHeaders);
+        } catch (Exception e) {
+            queryWizardStep3Response.setErrorMessage(e.getMessage());
+            return queryWizardStep3Response;
+        }
+        String queryId = createResponse.getResult();
+        CreateQuerySessionIDFilter.QUERY_ID.set(queryId);
+        queryWizardStep3Response.setQueryId(queryId);
+        GenericResponse<String> planResponse;
+        try {
+            planResponse = queryExecutor.plan(queryId);
+        } catch (Exception e) {
+            queryWizardStep3Response.setErrorMessage(e.getMessage());
+            return queryWizardStep3Response;
+        }
```

Comment:
When I ran this, I got a "Datawave could not generate a plan for the query" message when I tried ContentQuery logic type, which makes sense, because non-ShardQueryLogic types like ContentQuery typically won't have a planner involved. But in this case, you get stuck on the /showQueryWizardStep3 page with no way to retrieve the content result.

To replicate, do an EventQuery to pull back wikipedia records (e.g., "anarchy" unfielded should work, if you have the default data loaded), copy the RECORD_ID field's value (e.g., 20130305_0/enwiki/ibrtlu.qead3h.uz468c), then just plug that value into the query field for a ContentQuery

Reply:
I think in all cases you should have a Query ID though, so it'd be good to display that on the step3 page. E.g., its extremely handy to plug that value into a QueryMetricsQuery to troubleshoot queries as needed

Reply:
@keith-ratcliffe ,  initially this is working well and was tested mainly with EdgeQuery and EventQuery with the examples from the QuickStart guide since I have limited experience.

The queries I used were:
EventQuery    GENRES:action OR GENRES:adv*   (LUCENE)
EdgeQuery    SOURCE == 'kevin bacon' && TYPE == 'TV_COSTARS'   (JEXL) 

Those queries will get you some output for sure if you haven't seen any yet.  I will work on the suggestions you made but I have to leave this issue for a few days to get back on Accumulo .